docs/user_manual/dataset-terminology.md

```diff
@@ -10,23 +10,105 @@ Some terms regarding the data structures are explained here, including the defin
 
 ## Data structures
 
-- **Dataset:** Either a single or a batch dataset.
-    - **SingleDataset:** A data type storing input data (i.e. all elements of all components) for a single scenario.
-    - **BatchDataset:** A data type storing update and or output data for one or more scenarios. A batch dataset can contain sparse or dense data, depending on the component.
-- **DataArray** A data array can be a single or a batch array. It is a numpy structured array.     
-    - **SingleArray** A dictionary where the keys are the component types and the values are one-dimensional structured numpy arrays.
-    - **BatchArray:** An array of dictionaries where the keys are the component types and the values are two-dimensional structured numpy arrays.
-        - **DenseBatchArray:** A two-dimensional structured numpy array containing a list of components of the same type for each scenario.
-        - **SparseBatchArray:** A dictionary with a one-dimensional numpy int64 array and a one-dimensional structured numpy arrays. 
-
-### Type of Dataset 
-
-The types of `Dataset` include the following: `input`, `update`, `sym_output`, `asym_output`, and `sc_output`:
-Exemplery datasets attributes are given in a dataset containing a `line` component.
-
-- **input:** Contains attributes relevant to configuration of grid. 
+```{mermaid}
```

Comment:
can't we generate this directly from the dataset types in python? or do you think it's better to do it manually like this?

Reply:
We can have it via module diagrams: https://github.com/PowerGridModel/power-grid-model/issues/695
But only for this case via mermaid is only possible manually.